### PR TITLE
[README] add section on how to author types for ts >= 3.1

### LINKED ISSUE #31983: Using typesVerisons
In my latest pull request the Travis build complains with `Error: Don't use // TypeScript Version for newer TS versions, use typesVerisons instead.`

Unluckily I can't find anything about that 🤔  Could anyone please help me with that? Where and how can I put this information into?

### PR DESCRIPTION
**Pull request by BendingBender**
Update for `README` to explain how to setup types for TS > 3.1

Fixes #31983

### CODE REVIEW

**Comment by uniqueiniquity** [comment]
@rbuckton @DanielRosenwasser @RyanCavanaugh

File: README.md [hunk 1]
```diff
@@ -266,6 +266,41 @@ For an NPM package, `export =` is accurate if `node -p 'require("foo")'` is the
...
+You will need to use the `typesVersions` feature of TypeScript 3.1 and above. You can find a detailed explanation
+of this feature in the [official TypeScript documentation](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-1.html#version-selection-with-typesversions).
+
+Here's a short explanation to get you started:
+
+1. You'll have to add a `package.json` file to your package definition, with the following contents:
+
+```json
+{
+    "private": true,
```
Comment (RyanCavanaugh) [inline README.md]: I don't think we want this entry

Comment (rbuckton) [reply README.md]: Actually, I believe we do. This prevents someone from inadvertently trying to publish the package. Our types publisher ignores this field when it generates the actual package.json file.

Comment (BendingBender) [reply README.md]: Yeah, it doesn't work without this field. The linter complains with:
```
package.json should set `"private": true`
```

**Review by rbuckton — APPROVED** [review]

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**